tests/scale/README.md

```diff
@@ -0,0 +1,154 @@
+	Context("ScaleTest", func() {
+		// sd to store test state and iteration information
+		var  sd *DataHandle
+
+		// WrapUp will call to evaluate and output results on test's on os.Stdout.
+		// Additional outputs can be defined through `sd` API
+	AfterEach(func() {
+		if sd != nil {
+			sd.WrapUp()
+		}
+	})
+})
+```
+
+Initialize the body of the test, pretty standard from E2E framework semantics:
+```go
+It("Deploys OSM and scales traffic Splits indefinitely", func() {
+	// Install OSM, enable Grafana and Prometheus self-deployment
+	t := Td.GetOSMInstallOpts()
+	t.DeployGrafana = true
+	t.DeployPrometheus = true
+	Td.InstallOSM(t)
+
+	// Helpers to get OSM's install handlers, but arbitrary ones can be provided
+	pHandle := Td.GetOSMPrometheusaHandle()
+	gHandle := Td.GetOSMGrafanaHandle()
+
+	// This could/should be called on `Context`, subject to when are resources available
+	sd = NewDataHandle(pHandle, gHandle, GetTrackedResources(), GetSaveDashboards())	
+
+	// Repetitive scale loop
+	sd.Iterate(func() {
+		// Code goes here
+	}
+})
+```
+
+Tracked Resources are defined by labels, and they select the resources which are monitored during the test:
+```go
+func  GetTrackedResources() []TrackedLabel {
+	return []TrackedLabel{
+		{
+			Namespace: "some-namespace",
+			Label: metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": "osm-controller",
+				},
+			},
+		},
+		{
+			Namespace: "some-namespace",
+			Label: metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": "osm-prometheus",
+				},
+			},
+		},
+	}
+}
+```
+
+Similarly, Grafana dashboards can be saved upon test exit if Grafana is available. The dashboard prefix/name and the panel Id must be known:
+```go
+func  GetSaveDashboards() []GrafanaPanel {
+	return []GrafanaPanel{
+		{
+			Filename: "cpu",
+			Dashboard: "ABCDEF/mesh-cpu-and-mem",
+			Panel: 14,
+		},
+		{
+			Filename: "mem",
+			Dashboard: "ABCDEF/mesh-cpu-and-mem",
+			Panel: 12,
+		},
+	}
+}
+```
+
+## Results
+ Iteration results will display on test logs to keep an idea of progress.
+Only observed resources at this point in time are displayed.
+```
+-- Successfully completed iteration 8 - took 1m36.700005257s
++------------------------+--------------+---------+
+|        Resource        | CpuAvg (96s) | Mem RSS |
++------------------------+--------------+---------+
+| osm-cont..p2rb/osm-c.. |         0.44 | 160 MB  |
+| osm-prom..n8gt/prome.. |         0.19 | 1.2 GB  |
++------------------------+--------------+---------+
+```
+
+Upon `WrapUp` call time, a more comprehensive view of the evolution of resources is displayed at the end of the test, displaying both CPU avg for iteration and RSS footprint at the end, as well as relative increase to previous iteration.
+
+Note that this will show all possible resources that ever appeared during the test that matched any tracked label (so in case of restarted/crashed pods, all metrics and iterations present will also show):
```

Comment:
> so in case of restarted/crashed pods, all metrics and iterations present will also show

Is it worth keeping track of which iterations contain results that include failed/restarted pods?

Reply:
I think we should (in fact, we should bake it in the framework afterTest part maybe, and call it a fail if we see pods crashed)

However, currently a crashed pod will stop showing stats after a certain iteration and instead some other pod will be not showing stats till the same iteration, should be pretty visual on a table to understand one crashed and another one came up.